llvm/lib/Object/ArchiveWriter.cpp

```diff
@@ -807,6 +827,30 @@ computeMemberData(raw_ostream &StringTable, raw_ostream &SymNames,
     }
   }
 
+  if (SymMap) {
```

Comment:
Perhaps worth showing in a test that if symbols are not requested (i.e. `llvm-ar S ...`), no EC map is generated?

Reply:
It's generally tested by #82898. This case uses GNU format, which can't have EC symbols. I added one more test here as well.